lib/spack/spack/spec.py

```diff
@@ -989,7 +989,7 @@ def __init__(self, spec_like=None,
         self.architecture = None
         self.compiler = None
         self.external_path = None
-        self.external_module = None
+        self.external_modules = None
```

Comment:
(question) should this default to being an empty list?

Reply:
I wouldn't mind changing the code in that sense. I left it like it is now to be defensive in this rather large PR since `None` might be a sentinel that something was not set yet (and even in the previous case we could have used an empty string). Would it be fine with you to save this change for a minor refactor PR following this one? Or do you prefer to do it here?

Reply:
This can be handled later